src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,248 @@
+        Assertions.assertFalse(Storage.people.contains(fourthUser),
+                "User shouldn't be added to storage.");
+    }
+
+    @Test
+    void userPasswordLength_isOk() {
+        var actual = registrationService.register(firstUser).getLogin().length();
+
+        Assertions.assertTrue(MIN_LENGTH <= actual);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userPasswordLength_isNotOk() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(fifthUser));
+
+        var expected = "Password must be at least"
+                + MIN_LENGTH
+                + " current length"
+                + fifthUser.getPassword().length();
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(fifthUser),
+                "User with short password should not be added to storage.");
+    }
+
+    @Test
+    void userPassword_isNull() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(sixthUser));
+
+        var expected = "User password cannot be null";
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(sixthUser),
+                "User with null password should not be added to storage.");
+    }
+
+    @Test
+    void userAges_isOk() {
+        var actual = registrationService.register(firstUser).getAge();
+
+        Assertions.assertTrue(MIN_AGE <= actual);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userAge_isNotOk() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(seventhUser));
+
+        var expected = "User must be at least "
+                + MIN_AGE
+                + " years old.";
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(seventhUser),
+                "User with incorrect age should not be added to storage.");
+    }
+
+    @Test
+    void userAge_isNegative() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(eighthUser));
+
+        var expected = "Age cannot be negative";
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(eighthUser),
+                "User with incorrect age should not be added to storage.");
+    }
+
+    @Test
+    void userAge_isNull() {
```

Comment:
test method name should have 3 parts, first part - name of the tested method, second - conditions, third -expected result,please adhere to the suggested naming convention